handler/handler.go

```diff
@@ -47,3 +49,16 @@ func (q *NatsQueue) Queue(req *queue.Request) error {
 
 	return err
 }
+
+func (q *NatsQueue) reconnectClient(clientID, clusterID, natsURL string) nats.ConnHandler {
+	return func(c *nats.Conn) {
+		c.Close()
+		nc, err := stan.Connect(clusterID, clientID, stan.NatsURL(natsURL))
```

Comment:
See above. If you create the NATS connection, you should here close the stan connection (not the nats one), just for cleanup, then call stan.Connect() again and pass the `c` connection with the `stan.NatsConn()` option. (you would not need the `NatsURL()` option btw).